ykcapi/src/lib.rs

```diff
@@ -61,4 +66,52 @@ mod c_testing {
         Box::leak(bm);
         ret
     }
+
+    #[no_mangle]
+    pub extern "C" fn __yktrace_start_tracing(kind: usize) -> *mut ThreadTracer {
+        let kind: TracingKind = match kind {
+            SW_TRACING => TracingKind::SoftwareTracing,
+            HW_TRACING => TracingKind::HardwareTracing,
+            _ => panic!(),
+        };
+        Box::into_raw(Box::new(start_tracing(kind)))
+    }
+
+    #[no_mangle]
+    pub extern "C" fn __yktrace_stop_tracing(tt: *mut ThreadTracer) -> *mut IRTrace {
```

Comment:
This returns a `Box` which users are expected to pass repeatedly to `__yktrcae_irtrace_get` before finally calling `__yktrace_drop_irtrace` (I think)?

Reply:
That's the general gist, but users won't see this. It's for the testing code only. Users will use ykrt, which internally will just use the Rust code.